### ISSUE TITLE
The !cd command doesn't work normally.

### ISSUE THREAD

**Issue by ShuhuaGao** [googlecolab/colabtools#40]
When attempting to change directory with _!cd_ command, it didn't seem to work well.  A notebook to reproduce this issue can be found at https://colab.research.google.com/notebook#fileId=1C9x_lQreBVBUvIBOtJD0NHecD6UAHYv4

For your convenience, the screenshots are also attached. 
![image](https://user-images.githubusercontent.com/20141984/35552941-b963d5e8-05d0-11e8-900c-afe7fa555b29.png)

![image](https://user-images.githubusercontent.com/20141984/35552984-e2e2aee4-05d0-11e8-8702-694b000e455e.png)

**Comment by craigcitro** [comment]
tl;dr: this is a [known gotcha](http://ipython.readthedocs.io/en/stable/interactive/shell.html#directory-management).

Commands prefixed with `!` are run in a new subshell -- so you're starting a new shell, changing directories, and ... letting the shell process quit. (You can do things like `!cd /tmp && ls` to run a command in another directory.

By default, IPython enables [*automagics*](http://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-automagic), which lets you skip the leading `%`.

This means that the `cd` command is shorthand for `%cd` ([docs](http://ipython.readthedocs.io/en/stable/interactive/magics.html#magic-cd)), which changes the directory of the python process itself, not a child subprocess. That's why these changes stick.

I'm closing as WAI, but feel free to ask if that isn't clear.

**Comment by ShuhuaGao** [comment]
@craigcitro Thanks very much. I just knew that IPython can work as a shell, because I only used Jupyter notebook  in the Windows environment and never tried shell commands before. I thought _!cmd_ was a feature of Colab.

Anyway, got it. Thank you.

**Comment by nyck33** [comment]
@craigcitro  @ShuhuaGao 

can someone please clarify what to put instead of `!cd foo/bar` ?

**Comment by ziatdinovmax** [comment]
What about  %cd foo/bar ?

**Comment by nyck33** [comment]
Thanks, I should have reported back that I tried that and it worked :)

**Comment by Aidamina** [comment]
> @craigcitro @ShuhuaGao
> 
> can someone please clarify what to put instead of `!cd foo/bar` ?

`%cd foo/bar`

**Comment by Monster-Gaming-Studios** [comment]
Can somebody please tell me a solution of the code in "How to reproduce" section of issue #1094

**Comment by colaboratory-team** [comment]
@Monster-Gaming-Studios see the last comment on this issue above yours: instead of `!cd` you want `%cd`. The reason is explained earlier in this issue.

**Comment by mrgransky** [comment]
Instead of `! cd my_path/my_dir` please try `%cd my_path/my_dir`! 

To confirm, you could now try `! pwd` to ensure you're in new directory!

**Comment by utkarsh92** [comment]
Still not able to use **_environment variables_** with `cd`. Basically, I want to store a path as a variable so I don't have to type it again and again. For example:

![image](https://user-images.githubusercontent.com/10046302/120299345-8d138500-c2e8-11eb-9059-8887443df2da.png)
